### PR TITLE
Craig 261

### PR DESCRIPTION
**Pull request by jvallexm**
routing table / route

### CODE REVIEW

File: src/components/forms/RoutingTableRouteForm.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            value={this.state.next_hop}
+            labelText="Next Hop"
+            invalidCallback={() =>
+              isNullOrEmptyString(this.state.next_hop) ||
+              isIpv4CidrOrAddress(this.state.next_hop) === false
+            }
+            invalidText="Next hop must be a valid IP"
+            onChange={this.handleInputChange}
+            disabled={this.state.action !== "deliver"}
+            className="fieldWidthSmaller"
```
Comment (isaaceichelberger) [inline src/components/forms/RoutingTableRouteForm.js:105]: can a hop be a cidr block? invalid text makes me think no

File: src/components/forms/RoutingTableForm.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        <IcseFormGroup>
+          <IcseToggle
+            id={composedId + "-direct-link-toggle"}
+            labelText="Direct Link Ingress"
+            defaultToggled={this.state.route_direct_link_ingress}
+            onToggle={this.handleToggle}
+            tooltip={{
+              content:
+                "If set to true, the routing table is used to route traffic that originates from Direct Link to the VPC. To succeed, the VPC must not already have a routing table with the property set to true",
+            }}
```
Comment (Chris-Springstead) [inline src/components/forms/RoutingTableForm.js:82]: might need to add `align` and `alignModal` values so tooltip doesn't get cut off

Comment (jvallexm) [reply src/components/forms/RoutingTableForm.js:82]: what alignment ?

File: src/components/forms/RoutingTableRouteForm.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        <IcseFormGroup>
+          <IcseNameInput
+            componentName="routing-table-route"
+            id={this.props.data.name + "-route-name"}
+            hideHelperText
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() => {
+              this.props.invalidCallback(this.state, this.props);
+            }}
```
Comment (isaaceichelberger) [inline src/components/forms/RoutingTableRouteForm.js:53]: same as above, should be 
```js
invalid={this.props.invalidCallback(this.state, this.props}
```

File: src/components/forms/RoutingTableForm.js (hunk 2 above)
Comment (Chris-Springstead) [reply src/components/forms/RoutingTableForm.js:82]: `bottom-left` for both I believe

File: src/components/forms/RoutingTableForm.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  }
+
+  handleToggle(name) {
+    this.setState(this.toggleStateBoolean(name, this.state));
+  }
+  render() {
+    let composedId = this.props.data.name + "-route-form";
+    let innerFormProps = {
+      arrayParentName: this.props.data.name,
+      cluster: this.props.data,
```
Comment (isaaceichelberger) [inline src/components/forms/RoutingTableForm.js:43]: rename to route

File: src/components/forms/RoutingTableRouteForm.js (hunk 3 above)
Comment (jvallexm) [reply src/components/forms/RoutingTableRouteForm.js:53]: this was not working for me, needed to return a function

Comment (jvallexm) [reply src/components/forms/RoutingTableRouteForm.js:53]: nm

**Review by isaaceichelberger — APPROVED** [review]

**Pull request merged**